### PR TITLE
docs(jwt): update options config of `jwt` (#87)

### PR DESCRIPTION
**Pull request by lluia**
## Changes 💡

Following out update of `jose` https://github.com/nextauthjs/next-auth/pull/3039, the way to configure `jwt` changed. These are the options changes I inferred from the PR:
```diff
export default NextAuth({
  // ...
  jwt: {
    secret,
    maxAge,
-   encryptionKey
-   signingKey
-   encryptionKey
-   verificationOptions
    encode({
        token
        secret
        maxAge
-       signingKey
-       signingOptions
-       encryptionKey
-       encryptionOptions
-       encryption
    }) {},
    decode({
        token
        secret
-       maxAge
-       signingKey
-       verificationKey
-       verificationOptions
-       encryptionKey
-       decryptionKey
-       decryptionOptions
-       encryption
    }) {}
  }
})
```

I updated the relevant docs I could find mentioning `jwt` config in **v4**.  I lack a bit of context for this 🎻

## Affected issues 🎟

#87

## Screenshot (If Applicable) 📷

–

### CODE REVIEW

File: docs/configuration/options.md [hunk 1]
```diff
@@ -140,9 +140,9 @@ jwt: {
   // Defaults to `session.maxAge`.
   maxAge: 60 * 60 * 24 * 30,
   // You can define your own encode/decode functions for signing and encryption
-  // if you want to override the default behaviour.
+  // if you want to override the default behavior.
   async encode({ secret, token, maxAge }) {},
-  async decode({ secret, token, maxAge }) {},
+  async decode({ secret, token }) {},
```
Comment (lluia) [inline docs/configuration/options.md:145]: From the [PR in core](https://github.com/nextauthjs/next-auth/pull/3039/files#diff-405c2a1f8acb4c30e45523ca6c4e62dae1f83da7e1028080448de989dbc5e73d) I can &nbsp;👀 that `encode` has kept `maxAge` as a param but `decode` not. I got initially a bit suspicious about this, so highlighting it here...

File: docs/errors.md [hunk 2]
```diff
@@ -129,17 +129,6 @@ The algorithm used for generating your key isn't listed as supported. You can ge
...
-
-```
-  jwt: {
-    signingKey: {"kty":"oct","kid":"--","alg":"HS256","k":"--"},
-    verificationOptions: {
-      algorithms: ["HS256"]
-    }
-  }
-```
-
```
Comment (lluia) [inline docs/errors.md:142]: Configuring it like this is not longer possible, however I believe you could achieve this through the `encode` function? Something like:
```ts
  jwt: {
    secret: process.env.SECRET,
    encode: async ({ secret, token }) => {
      return jwt.sign({...token, userId: token.id}, secret, {
        algorithm: "HS256",
        kty: "oct",
        kid: "––",
        k: "––"
      });
    },
```

File: docs/faq.md [hunk 3]
```diff
@@ -347,7 +347,7 @@ Using explicit public/private keys for signing is strongly recommended.
 
 <details>
 <summary>
-  <h3 style={{display:"inline-block"}}>What signing and encryption standards does NextAuth.js support?</h3>
```
Comment (lluia) [inline docs/faq.md:350]: I removed this inline style as it doesn't seem to have any effect on the layout 📐

File: docs/configuration/options.md (hunk 1 above)
Comment (balazsorban44) [reply docs/configuration/options.md:145]: Correct. There is no point in passing `maxAge` when decoding. You are just parsing (decoding) the token value.

File: docs/errors.md (hunk 2 above)
Comment (balazsorban44) [reply docs/errors.md:142]: Correct. `jwt.sign` has to be imported from something like `jose` or `jsonwebtoken`.

I think we can safely drop this section tbh, we are not recommending for anyone to use the NextAuth.js issued JWT outside of their app, so these changes are only necessary in very rare cases.

We generally don't want people to even use `encode`/`decode`, it's there for heavy users who hopefully understand its purpose.

**Pull request merged**